### PR TITLE
feat(ST-84): add classroom card on the dashboard for teacher only

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
## Description of Change
- Added ClassroomCard.tsx component with types file.
- Added ClassroomCardList.tsx file where ClassroomCard is getting called.
- Updated the Navbar.tsx, as user's changes profile the data is getting cached, so no update happens when another user logs in
- Updated the DashboardContainer file, now it's dynamic when a teacher have already created classrooms.
- Added the card image to the public folder.
- Updated the classroom.api.ts file, used tags so that user does not need to reload the page for seeing the changes.

## Associated Ticket Link(s)

- https://trello.com/c/iRAPGs9b

## Screenshots / Screen Recordings

https://github.com/user-attachments/assets/d286cb66-3b9d-4eb0-8682-e4add4ef8cd5

### CODE REVIEW

File: modules/Dasboard/components/ClassroomCardList/ClassroomCardList.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import { useGetClassroomsQuery } from "@/shared/redux/rtk-apis/classrooms/classrooms.api";
+
+import ClassroomCard from "../ClassroomCard/ClassroomCard";
+import { Classroom } from "./ClassroomCardList.types";
+
+const ClassroomList = () => {
+  const { data: classrooms, isLoading, error } = useGetClassroomsQuery();
+
+  if (isLoading) return <p>Loading classrooms...</p>;
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/components/ClassroomCardList/ClassroomCardList.tsx:11]: Create a shared loading spinner component and use that for these loading states

File: modules/Dasboard/components/ClassroomCardList/ClassroomCardList.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  if (error) return <p>Error loading classrooms</p>;
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/components/ClassroomCardList/ClassroomCardList.tsx:12]: Show a toast here

File: modules/Dasboard/components/ClassroomCardList/ClassroomCardList.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  if (!classrooms || classrooms.length === 0) return null;
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/components/ClassroomCardList/ClassroomCardList.tsx:13]: Not sure if I understand what's happening here. If there are no classrooms, would the users see an empty screen?

File: modules/Dasboard/components/Navbar/Navbar.tsx [hunk 4]
```diff
@@ -20,7 +22,12 @@ const Navbar = () => {
   const router = useRouter();
   const [isModalOpen, setIsModalOpen] = useState(false);
 
+  const logout = () => (dispatch: TAppDispatch) => {
+    dispatch(projectApi.util.resetApiState());
+  };
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/components/Navbar/Navbar.tsx:27]: Is it necessary to wrap this by a method? Can't this be directly invoked inside the handleLogin method?

File: modules/Dasboard/components/ClassroomCardList/ClassroomCardList.tsx (hunk 3 above)
Comment (akib-raihan-sazim) [reply modules/Dasboard/components/ClassroomCardList/ClassroomCardList.tsx:13]: No bhaiya, the user will see the CreateClassroomButton, but my implementation was wrong here, I have updated my code.
Initially I was fetching data from this component which is not needed, I could just fetch the classroom data from the dashboard and pass the data in this component.

File: modules/Dasboard/components/Navbar/Navbar.tsx (hunk 4 above)
Comment (akib-raihan-sazim) [reply modules/Dasboard/components/Navbar/Navbar.tsx:27]: It can be done.

File: modules/Dasboard/containers/teacher/DashboardContainer.tsx [hunk 5]
```diff
@@ -1,13 +1,66 @@
...
+
+    if (error && hasAttemptedLoad) {
+      timer = setTimeout(() => {
+        if (!classrooms) {
+          notifications.show({
+            title: "Error",
+            message: "Failed to load classrooms. Please try again later.",
+            color: "red",
+            autoClose: 5000,
+          });
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/containers/teacher/DashboardContainer.tsx:33]: You don't need to wrap all of this or the effect around this method. Just show this toast when error is true

File: modules/Dasboard/containers/teacher/DashboardContainer.tsx [hunk 6]
```diff
@@ -1,13 +1,66 @@
...
 
-const DashboardContainer = () => (
-  <div>
-    <Navbar />
-    <CreateClassroomButton />
-  </div>
-);
+const DashboardContainer: React.FC = () => {
+  const { data: classrooms, isLoading, error } = useGetClassroomsQuery();
+  const [hasAttemptedLoad, setHasAttemptedLoad] = useState(false);
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/containers/teacher/DashboardContainer.tsx:14]: You probably don't need this

File: shared/redux/rtk-apis/api.config.ts [hunk 7]
```diff
@@ -5,8 +5,12 @@ import baseQuery from "@/shared/redux/rtk-apis/baseQuery";
...
   baseQuery,
-  tagTypes: [],
+  tagTypes: ["Classrooms"],
   endpoints: () => ({}),
 });
 
-export default projectApi;
+export const enhancedProjectApi = projectApi.enhanceEndpoints({
+  addTagTypes: ["Classrooms"],
+});
```
Comment (ishmam-iqbal-sazim) [inline shared/redux/rtk-apis/api.config.ts:14]: You don't need this. Providing it in tagTypes above should be fine

File: modules/Dasboard/containers/teacher/DashboardContainer.tsx [hunk 8]
```diff
@@ -1,13 +1,48 @@
...
+  const {
+    data: classrooms,
+    isLoading,
+    error,
+  } = useGetClassroomsQuery(undefined, {
+    skip: user.userId === null,
+  });
+
+  if (error) {
+    console.log(user);
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/containers/teacher/DashboardContainer.tsx:23]: let's remove unnecessary console logs

File: modules/Dasboard/containers/teacher/DashboardContainer.tsx [hunk 9]
```diff
@@ -1,13 +1,48 @@
...
-  </div>
-);
+const DashboardContainer: React.FC = () => {
+  const user = useAppSelector(selectAuthenticatedUser);
+  const {
+    data: classrooms,
+    isLoading,
+    error,
+  } = useGetClassroomsQuery(undefined, {
+    skip: user.userId === null,
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/containers/teacher/DashboardContainer.tsx:19]:
```suggestion
    skip: !user.userId,
```

[NIT]

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**